packages/opentelemetry-metrics/src/MetricObservable.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+
+type Subscriber = (value?: number) => void;
+
+/**
+ * Implements the Metric Observable pattern
+ */
+export class MetricObservable implements api.MetricObservable {
+  private _subscribers: Subscriber[] = [];
+
+  next(value: number) {
+    for (const subscriber of this._subscribers) {
+      subscriber(value);
```

Comment:
Can we use `setImmediate(subscriber, value)` to avoid calling subscribers synchronously

Reply:
Why do think it would be a problem if we call them synchronously. The common pattern for such things is to make it synchronous. Also if user is calling "next" I would assume that this is the moment it should be updated, including the time as well, using setImmediate the time won't be accurate. I also think that this might cause some unexpected behaviour, if for any reason you would like to also call "collect" after you update values. I would be rather against that.

Reply:
Just to avoid blocking the event loop for too long if there are many subscribers to update, or if the subscribers take too long to process a value. A user may also `subscribe` to updates, not just us, and the user is unlikely to know their callback is in a performance sensitive context

Reply:
The "hidden" async operation will prevent user from calling `collect` immediately after updating the values, and time for updating the value will also be different then moment of calling update. Besides that "blocking loop" would last as long as callback in "standard way" anyway.

Reply:
@mayurkale22 @vmarchaud whats your thoughts on that ?

Reply:
I agree with @dyladan, the spec state that the update path should be as fast as possible so i would prefer using `setImmediate` here.

Reply:
@vmarchaud setImmediate will delay update - it won't be updated asap as spec says

Reply:
I don't find any requirement on specific time requirement, could you give a link please ?

Reply:
@vmarchaud you wrote `the spec state that the update path should be as fast as possible ` so setImmediate won't update asap but it will update in next cycle

Reply:
I see, the whole spec state:
```
Metric updates made via a bound instrument, when used with an Aggregator defined by simple atomic operations, should follow a very short code path
```
I interpret `short code path` from a performance requirement and not as a time requirement. So using `setImmediate` would be a better solution for me

Reply:
I'm strongly against it but if majority want's setImmediate with it's all consequences fine for me, @mayurkale22 you also vote for setImmediate ?